mesh/mesh.cpp

```diff
@@ -5088,6 +5091,21 @@ int *Mesh::CartesianPartitioning(int nxyz[])
 int *Mesh::GeneratePartitioning(int nparts, int part_method)
 {
 #ifdef MFEM_USE_METIS
+
+   int print_messages = 1;
+   // If running in parallel, print messages only from rank 0.
+#ifdef MFEM_USE_MPI
+   int init_flag, fin_flag;
+   MPI_Initialized(&init_flag);
+   MPI_Finalized(&fin_flag);
+   if (init_flag && !fin_flag)
+   {
+      int rank;
+      MPI_Comm_rank(GetGlobalMPI_Comm(), &rank);
```

Comment:
Unlikely scenario: Someone wants to create a `ParMesh` on a communicator that does not include the "global" MPI_Comm (maybe process 0 is doing IO or something). Then this method won't print messages, but the intent is that it should.

A possible fix is to make the `print_messages` flag an argument to the method. The only callers are the `ParMesh` constructor (which has a communicator and can make the same decision that is made here) and `mesh_explorer`, which can just tell this method to always print messages.

As I said, the issue here is minor and unlikely to arise in practice, so I'm not going to insist on this.

Reply:
Yes, you are right! I also though about this issue but decided it is not really worth addressing at this point.